pam/return.go

```diff
@@ -1,54 +1,41 @@
 package main
 
+import (
+	"github.com/msteinert/pam"
+)
+
 // Various signalling return messaging to PAM.
 
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
+type pamReturnStatus interface {
+	Status() pam.StatusError
+	Error() string
+}
+
+// TODO: Remove this when we've tests coverage that ensures this is the case.
+// ensure that [pam.NewTransactionError] implements [pamReturnStatus].
+var _ pamReturnStatus = pam.NewTransactionError(pam.ErrAbort, nil)
+
+// pamStatus signals PAM module to return with provided pam.Success and Quit tea.Model.
 type pamSuccess struct {
+	pam.TransactionError
```

Comment:
Why adding this field? It’s not used at all. I know you want to mirror the error pam returned type, but this is cluttering and confusing the construction of the resulting type.

Reply:
Hence the fact that we should not rely on the interface IMHO, implementing something empty just to implement an interface is a sign that something is wrong in the approach and is not go-ish